### PR TITLE
Removed request dependency

### LINKED ISSUE #9: [hotfix] Fix promotion Twitch API
Currently the implementation of the twitch api doesn't seem to work. Needs to be fixed.

Could use a Twitch Api wrapper or npm module. Please refrain from using the npm `requests` module, if possible.

### PR DESCRIPTION
**Pull request by notfooddev**
Replaced request dependency with node-fetch. I also noticed no oauthtoken was required for the request, which is a requirement according to the Twitch API, so I added it to the config.

This change has been tested.

### CODE REVIEW

File: src/events/presenceUpdate.ts [hunk 1]
```diff
@@ -55,39 +55,40 @@ export default async function PresenceUpdateEvent(discordClient: DiscordClient,
...
+			method: 'get',
 			headers: {
 				'Client-ID': config.twitch.clientId,
+				'User-Agent': userAgent,
+				'Authorization': 'Bearer ' + config.twitch.oauthToken
 			}
-		}, function (error, res, body) {
-			if (error || !body) {
+		})
+        .then(res => res.json())
```
Comment (WesVleuten) [inline src/events/presenceUpdate.ts]: Mixed indents

**Review by WesVleuten — CHANGES_REQUESTED** [review]
- Mixed indents
- I think we should be using an application token instead of an arbitrary auth token
- `then` -> `await`

**Comment by notfooddev** [comment]
image

Normally it's not black, this stream just had a black screen.

File: config.example.json [hunk 2]
```diff
@@ -1,7 +1,8 @@
 {
 	"botOwnerUserId": "",
 	"twitch": {
-		"clientId": ""
+		"clientId": "",
+		"accessToken": ""
```
Comment (WesVleuten) [inline config.example.json]: Not a fan of using an arbitrary accessToken. We can use Twitch's App access tokens to fully automate the tokens.

[App access tokens](https://dev.twitch.tv/docs/authentication#types-of-tokens)
[OAuth Client Credentials Flow](https://dev.twitch.tv/docs/authentication/getting-tokens-oauth#oauth-client-credentials-flow)

File: src/events/presenceUpdate.ts [hunk 3]
```diff
@@ -55,39 +55,36 @@ export default async function PresenceUpdateEvent(discordClient: DiscordClient,
...
+			method: 'get',
 			headers: {
 				'Client-ID': config.twitch.clientId,
+				'User-Agent': userAgent,
+				'Authorization': 'Bearer ' + config.twitch.accessToken
 			}
-		}, function (error, res, body) {
-			if (error || !body) {
+		})
+		.then(res => res.json())
```
Comment (WesVleuten) [inline src/events/presenceUpdate.ts]: Awaits > Thens

My guess:
```
const request = await fetch(...);
const data = await request.json();
```

File: src/events/presenceUpdate.ts [hunk 4]
```diff
@@ -55,39 +55,36 @@ export default async function PresenceUpdateEvent(discordClient: DiscordClient,
...
-				.setAuthor(`${guildMember.user.tag}`, `${guildMember.user.avatarURL}`)
-				.setDescription(`**Streamer:** ${strem.user_name}`)
-				.addField("**Stream Title:**", `${strem.title}`, false)
+				.setImage(thumbnail)
+				.setAuthor(`${guildMember.displayName}`, `${guildMember.user.displayAvatarURL()}`)
+				.setDescription(`**Streamer:** ${stream.user_name}`)
+				.addField("**Stream Title:**", `${stream.title}`, false)
 				.addField("**Stream URL:**", `${streamUrl}`, false);
-
+
```
Comment (WesVleuten) [inline src/events/presenceUpdate.ts:114]: Weird left in indents

**Review by WesVleuten — CHANGES_REQUESTED** [review]
Looking good, some small points of possible improvement.

Also, we should move the twitch function to a more generalized library. This could also be done at a later time, your choice.

**Review by WesVleuten — APPROVED** [review]

**Pull request merged**